### ISSUE TITLE
Error "The broker does not support DESCRIBE_CONFIGS"

### ISSUE THREAD

**Issue by yorek** [Azure/azure-event-hubs-for-kafka#61]
Description
===========
When using Debezium, which is a Kafka Connect connector, I get the error "The broker does not support DESCRIBE_CONFIGS", when trying to use Event Hub to store schema changes. See Issue #53 which has been closed but, though a workaround (not saving the schema changes) has been found, the problem is still there and prevents a full debezium usage.

How to reproduce
================
Setup Debezium to use EventHub instead of Kafka

Has it worked previously?
=========================
No

Checklist
=========

**IMPORTANT**: We will close issues where the checklist has not been completed.

Please provide the following information:
 
 - [x] Verified that port 9093 is not blocked by firewall
 - [x] Verified the namespace is either Standard or Dedicated tier (i.e. it is not Basic tier, which isn't supported)
 - [x] Sample you're having trouble with: `Debezium Tutorial`
 - [x] Apache Kafka version: `2.2.0`
 - [x] Kafka client configuration: `none`
 - [x] Namespace and EventHub/topic name
 - [x] Producer failure

**Comment by arerlend** [comment]
Error message is accurate, DESCRIBE_CONFIGS is in fact not supported.  We're aware that it's a blocker for some Kafka applications (not just for the Debezium connector) - it will be implemented in the near future.  Note to readers - DESCRIBE_CONFIGS is not needed for base Kafka functionality.

**Comment by yorek** [comment]
Thanks a lot Arthur. Any timeframe that you can share? Thanks!

**Comment by arerlend** [comment]
No exact timeframe I can give you now.  We're aware it's heavily utilized by a large portion of Kafka extensions so it's relatively high priority on the roadmap.

**Comment by arerlend** [comment]
@yorek DescribeConfigs is being investigated now.

@conorr thanks for the contribution on the other repo.  If you've got the time I'd love to get a sample or working config, and I'll post it on this repo.

**Comment by yorek** [comment]
Yeah! :)

**Comment by conorr** [comment]
No problem. Do you mean a sample for running kafdrop UI against Event Hubs for Kafka? Or a sample consumer? I can provide either.

**Comment by arerlend** [comment]
@conorr I was thinking the kafdrop UI config.  I'll be adding a repo folder with configs for management tooling and this looks like an interesting addition.

**Comment by conorr** [comment]
Save config to `kafka.properties`:

```
security.protocol=SASL_SSL
sasl.mechanism=PLAIN
sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="$ConnectionString" password="<connection-string-goes-here>";
```

Pull kafdrop docker image (the `DESCRIBE_CONFIGS` fix is not yet in the `latest` tag, but I imagine it will be soon):

```bash
docker pull obsidiandynamics/kafdrop:3.22.0-SNAPSHOT
```

Run the container, referencing the `kafka.properties` file:

```bash
docker run -d --rm -p 9000:9000 \
    -e KAFKA_BROKERCONNECT=your-event-hubs-namespace.servicebus.windows.net:9093 \
    -e KAFKA_PROPERTIES=$(cat kafka.properties | base64) \
    obsidiandynamics/kafdrop:3.22.0-SNAPSHOT
```

This will run kafdrop UI at http://localhost:9000.

**Comment by arerlend** [comment]
We've added a handler for DescribeConfigs.  I'll try out kafdrop now.

**Comment by conorr** [comment]
I confirm the DescribeConfigs fix. I used an older version of the kafdrop docker image that didn't have my workaround fix. Thank you!

**Comment by arerlend** [comment]
Thanks @conorr.  Closing for now, but pinged @yorek to see if he can try out his scenario as well.

For future readers - please open a new issue if you're seeing exceptions in DescribeConfigs paths.

**Comment by arerlend** [comment]
Broker-level configs need support.

**Comment by ywchang** [comment]
Hi @arerlend, can I understand the current state of this issue? I was using the same kafka connect plugin `debezium/connect` and I am currently having issue with event hub.

The error looks like this, and I high suspect it fails when trying to get broker config from calling adminclient describeConfigs method.

```

**Comment by ywchang** [comment]
Hi @arerlend

Good day!

Is there any update for this issue?

**Comment by arerlend** [comment]
@ywchang we haven't implemented adding broker-level configs to the configs yet.  Is this blocking a required production scenario for you?

**Comment by ywchang** [comment]
Hi @arerlend

For me, the scenario is that we are using Debezium Kafka Connector with Azure Event Hub to extract data from a external database to Event Hub in production environment. The Debezium will query the broker config to get a replication settings before using its default value.

Since the Event Hub hasn't supported this query, an exception will be thrown using the official debezium. I have to fork the repo and do some code changes to make it work through. So just wondering if there is a timeline for this so that we can avoid changing the source code.

**Comment by Adpersonam** [comment]
Hi @arerlend,
Any update on this issue?

**Comment by ssugar** [comment]
Hi @arerlend - any recent update on this?  I receive the same "org.apache.kafka.connect.errors.ConnectException: Creation of database history topic failed, please create the topic manually" error when connecting Debezium (Kafka-Connect) in Azure Event Hubs.  

There is a workaround listed in #53 which is to add the following to the connector configuration:
"database.history":"io.debezium.relational.history.MemoryDatabaseHistory"
My understanding is that this would store the database history in memory, which probably won't survive a restart.  During some research, it looks like it may be possible to switch the above to:
"database.history":"io.debezium.relational.history.FileDatabaseHistory",
"database.history.file.filename"="history.dat"
And I'm assuming that the database history would then be able to survive a restart. (still need to do some testing).

Overall, it would be ideal if no additional workaround was required. 

Anyone have any thoughts on this?

**Comment by yehudamakarov** [comment]
@ssugar, what are you asking? if it is ideal that azure not completely support the kafka endpoint, because we can use 

    "database.history":"io.debezium.relational.history.FileDatabaseHistory",
    "database.history.file.filename":"history.dat"

**Comment by ssugar** [comment]
@yehudamakarov - I wasn't asking anything, was just saying it would be ideal if EventHubs provided a kafka endpoint that didn't require any workaround

**Comment by yehudamakarov** [comment]
Ah!! :) ok great we are definitely on the same page!

**Comment by alastairs** [comment]
@yehudamakarov @arerlend Is there any update available on whether Azure Event Hubs now supports broker-level configs for DESCRIBE_CONFIGS requests? I'm trying to determine whether or not [the workaround for Debezium](https://github.com/Azure-Samples/azure-sql-db-change-stream-debezium/blob/master/documentation/SQL-Server-Connector-Configuration-Value.md#database-history) is still required.
